### PR TITLE
Kafka integration - bug fix and enhancements

### PR DESCRIPTION
**Pull request by Itay4**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19482 https://github.com/demisto/etc/issues/18743

## Required version of Demisto
Any

## Does it break backward compatibility?
   - Yes
       - Further details: we changed the fetch logic 

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] Code Review

## Dependencies
None

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](https://github.com/demisto/content/blob/6b326dbf6eda33d5d87601486f103f01a0ab1289/Integrations/Kafka_V2/Kafka_V2.yml)
- [ ] [CHANGELOG](https://github.com/demisto/content/blob/6b326dbf6eda33d5d87601486f103f01a0ab1289/Integrations/Kafka_V2/CHANGELOG.md)

### CODE REVIEW

File: Integrations/Kafka_V2/Kafka_V2.py [hunk 1]
```diff
@@ -316,75 +332,121 @@ def fetch_partitions():
...
+        max_messages = 50
+
+    last_fetched_partitions_offset = json.loads(demisto.getLastRun().get('last_fetched_partitions_offset', '{}'))
+
+    incidents = []
+
+    message_counter = 0
+
+    for topic in client.topics.values():
+        if (topics_to_fetch_from and topic.name in topics_to_fetch_from) or not topics_to_fetch_from:
```
Comment (glicht) [inline Integrations/Kafka_V2/Kafka_V2.py]: If I understand correctly if topic_to_fetch_from is empty we fetch from all topics. Sounds to me extreme. Let's always require at least one topic defined.

File: Integrations/Kafka_V2/Kafka_V2.py [hunk 2]
```diff
@@ -316,75 +332,123 @@ def fetch_partitions():
...
+    except ValueError:
+        max_messages = 50
+
+    last_fetched_partitions_offset = json.loads(demisto.getLastRun().get('last_fetched_partitions_offset', '{}'))
+
+    incidents = []
+
+    message_counter = 0
+
+    for topic in client.topics.values():
```
Comment (Itay4) [inline Integrations/Kafka_V2/Kafka_V2.py]: 1 topic

File: Integrations/Kafka_V2/Kafka_V2.py [hunk 3]
```diff
@@ -316,75 +332,123 @@ def fetch_partitions():
...
+                partition_id = str(partition.id)
+                if (partition_to_fetch_from and partition_id in partition_to_fetch_from) or not partition_to_fetch_from:
+                    offset = int(last_fetched_partitions_offset.get(partition_id, offset_to_fetch_from))
+                    max_offset = offset
+
+                    consumer = topic.get_simple_consumer(
+                        auto_offset_reset=offset,
+                        reset_offset_on_start=True,
+                        consumer_timeout_ms=5000,
+                        partitions=[partition]
```
Comment (Itay4) [inline Integrations/Kafka_V2/Kafka_V2.py]: array of partitions and reset offsets

File: Integrations/Kafka_V2/Kafka_V2.py [hunk 4]
```diff
@@ -316,75 +332,123 @@ def fetch_partitions():
...
-    if os.path.isfile('client_key.key'):
-        os.remove(os.path.abspath('client_key.key'))
+def main():
+    LOG('Command being called is {0}'.format(demisto.command()))
+    global log_stream
+    try:
+        start_logging()
+
+        # Initialize KafkaClient
+        if VERIFY_SSL:
```
Comment (Itay4) [inline Integrations/Kafka_V2/Kafka_V2.py]: new param

**Comment by kirbles19** [comment]
@Itay4 Done. Updated the changelog. Please add periods "." to the end of the descriptions for each of the new outputs.

**Pull request merged**